cask.el

```diff
@@ -806,6 +810,15 @@ url to the fetcher source."
         (setf (cask-dependency-ref dependency) ref))
       (-when-let (branch (plist-get args :branch))
         (setf (cask-dependency-branch dependency) branch)))
+    (-when-let (archive (plist-get args :archive))
+      (if (not (boundp 'package-pinned-packages))
+          (princ "Archive pinning is only supported with Emacs v24.4+.\n")
```

Comment:
My approach was to use `(princ ...)`  to warn the user, not to send an error and stop the execution.
What do you think about it? Is it better to exit with an error or warn the user, ignore the archive parameter and continue?